### DISCUSSION TITLE
Privacy Issue - regarding default config toml use in the Ubuntu Instalation

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#1856] [Potential issues]
Note: Before starting with the issue I want to ask you to understand I am not a good writter, neither this is my first lenguage. Please be more understandable if I do not use the propeer grammar or words to address the issue.

      Hello, I belive there is an underling issue regarding security in the preset config file (dnscrypt-proxy.toml) of the ubuntu repository package of dnscrypt-proxy. The config file pre-installed on my computer default option for server_names is set to ['cloudflare'] which represent an issue since cloudflare history of privacy related actions is not good, since they already control a big chunk of the web traffic. I don't what to be spread any misinformation but is clear claudflare is at least not to be trusted. Given the reputation of ubuntu for been private and defend the right for free software and freedom, I do not understand why the default option for the servers look up in the ubuntu installation (I have not tested it on other systems) is cloudflare which is not very privacy respecting. 
     
     Q1-Why did the devs did not leave it blank? To make dnscript-proxy use other resolvers too.
     Q2-If you do not have anything to do with the way they put the options in the ubuntu repository build by default can you point me to how can I contact the responsables for those actions. 

Thank you for your patient, hope you have a good day.  

     System Information:
Linux ubuntu 5.4.0-1042-raspi #46-Ubuntu SMP PREEMPT Fri Jul 30 00:35:40 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux.
     Dnscrypt-proxy version installed is 2.0.31.

     
This is how the config file looks by default:
# Empty listen_addresses to use systemd socket activation
listen_addresses = []
server_names = ['cloudflare']

[query_log]
  file = '/var/log/dnscrypt-proxy/query.log'

[nx_log]
  file = '/var/log/dnscrypt-proxy/nx.log'

[sources]
  [sources.'public-resolvers']
  url = 'https://download.dnscrypt.info/resolvers-list/v2/public-resolvers.md'
  cache_file = '/var/cache/dnscrypt-proxy/public-resolvers.md'
  minisign_key = 'RWQf6LRCGA9i53mlYecO4IzT51TGPpvWucNSCh1CBM0QTaLn73Y7GFO3'
  refresh_delay = 72
  prefix = '

**Comment by ghost** [comment]
also I'm sorry for not using corectly the formating, I haven't use it enough

**Comment by jedisct1** [comment]
The default configuration shipped by this project doesn't use any specific resolver.

Installing the application by following the documentation provided here doesn't set a specific resolver either. Resolvers are automatically chosen according to the filters you define and how fast they are for your connection.

Maybe Ubuntu decided to ship a package with a different configuration. But this package is made by them. So it should be discussed on Ubuntu lists, forum or whatever is relevant to discuss Ubuntu packages. Nothing can be done here.

That being said, people have different opinions about Cloudflare and other big and small actors. And changing the configuration should be easy enough.

**Comment by ghost** [comment]
@jedisct1
Thanks for the answer, I will find the proper chat to post my consers.  I just have a question while changing the option in the config file I notice some of the options on the example-dnscrypt-proxy.toml file give errors when I tryed them on the same ubuntu instalation. Can you point to to a source to learn about how to customize the config file for the ubuntu installation (from the repository)?
